pkg/publicapi/apimodels/error.go

```diff
@@ -67,31 +66,37 @@ func (e *APIError) Error() string {
 }
 
 // Parse HTTP Resposne to APIError
-func FromHttpResponse(resp *http.Response) (*APIError, error) {
-
+func GenerateAPIErrorFromHTTPResponse(resp *http.Response) *APIError {
 	if resp == nil {
-		return nil, errors.New("response is nil, cannot be unmarsheld to APIError")
+		return NewAPIError(0, "API call error, invalid response")
 	}
 
 	defer resp.Body.Close()
 
 	body, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return nil, fmt.Errorf("error reading response body: %w", err)
+		return NewAPIError(
+			resp.StatusCode,
+			fmt.Sprintf("Unable to read API response body. Error: '%s'", err.Error()))
```

Comment:
nit: we can use `%q` here (and elsewhere) to add double quotes around the string and safely escapes any special characters. i.e.
```suggestion
			fmt.Sprintf("Unable to read API response body. Error: %q", err.Error()))
```

Reply:
Thanks for the suggestion. Code changed.